### PR TITLE
mesenhanced: add "enhanced" features similar to OSB.

### PR DESCRIPTION
**Pull request by Ganom**
closes #1632

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	{
+		if (tick)
+		{
+			tick = false;
+		}
+	}
+
+	private void onAnimationChanged(AnimationChanged event)
+	{
+		if (event.getActor() != client.getLocalPlayer() || event.getActor().getAnimation() != 3705 || !bones)
```
Comment (SRLJustin) [inline runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: should 3705 be replaced with ``AnimationID.USING_GILDED_ALTAR`` ?

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+			return;
+		}
+
+		event.getActor().setActionFrame(0);
+		event.getActor().setAnimation(GOBLIN_SALUTE);
+	}
+
+	private void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!this.leftClickLog && !this.quickBone)
```
Comment (SRLJustin) [inline runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: shouldn't it be ``||`` and not``&&`` ?

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+			&& event.getTarget().toLowerCase().contains("altar"))
+		{
+			event.getMenuEntry().setOption(QUICK_BONE);
+			event.setWasModified(true);
+		}
+	}
+
+	private void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!this.leftClickLog && !this.quickBone)
```
Comment (SRLJustin) [inline runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: shouldn't it be ``||`` and not``&&`` ?

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			client.setSelectedItemWidget(WidgetInfo.INVENTORY.getId());
+			client.setSelectedItemSlot(bonesIdx);
+			client.setSelectedItemID(bonesId);
+			tick = true;
+		}
+	}
+
+	private void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (!this.leftClickLog && !this.quickBone)
```
Comment (SRLJustin) [inline runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: shouldn't it be ``||`` and not``&&`` ?

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java (hunk 1 above)
Comment (SRLJustin) [inline runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: should 3705 be replaced with AnimationID.USING_GILDED_ALTAR ?

Comment (Lucwousin) [reply runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: Yes

File: runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java (hunk 2 above)
Comment (Lucwousin) [reply runelite-client/src/main/java/net/runelite/client/plugins/mesenhanced/MesEnhanced.java]: no, as it's a guard clause and you want to pass it if either one of them is not true, not only when both are

**Pull request merged**